### PR TITLE
Fix building with BV_{SYN,TUR} build args

### PR DESCRIPTION
**Pull request by williamboman**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -65,7 +65,7 @@ RUN chmod a+x /start.sh \
     ; \
     curl -L https://github.com/matrix-org/synapse/archive/$BV_SYN.zip -o s.zip \
     && unzip s.zip \
-    && cd /synapse-$BV_SYN \
+    && cd /synapse-$(echo $BV_SYN | tr -d v) \
```
Comment (williamboman) [inline Dockerfile:68]: This is pretty ugly but does the trick. Won't break regular SHAs either as v is not a hexadecimal character ¯\_(ツ)_/¯

**Comment by silvio** [comment]
Hi, thanks for your contribution.

is `ARG` supported from http://hub.docker.com? That's why I have changed my first approach with `ARG` to `ENV`. (commit: 17b2b24ec37cf405e47d768cc6da14e544c3c8b9)

**Comment by williamboman** [comment]
> is ARG supported from http://hub.docker.com? That's why I have changed my first approach with ARG to ENV. (commit: 17b2b24)

Oh, doesn't look like it - https://github.com/docker/hub-feedback/issues/460.

But I have [other automated builds](https://hub.docker.com/r/thelounge/lounge/~/dockerfile/) on DockerHub that uses `ARG` directives supplied with default values and they seem to build fine :confused:? I think the issue is being able to specify custom build arguments through DockerHub. I suspect DockerHub does something like `s/ARG/ENV/g`.

**Comment by silvio** [comment]
I don't want it in the sources, at least the hub has no native support for the`ARG` thing.

**Comment by williamboman** [comment]
Like so? 

``` diff
diff --git a/Dockerfile b/Dockerfile
index ba5907f..a607ed1 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -13,8 +13,8 @@ EXPOSE 8448
 VOLUME ["/data"]

 # Git branch to build from
-ARG BV_SYN=master
-ARG BV_TUR=master
+ENV BV_SYN=master
+ENV BV_TUR=master
 # https://github.com/python-pillow/Pillow/issues/1763
 ENV LIBRARY_PATH=/lib:/usr/lib

diff --git a/readme.md b/readme.md
index 5e752e5..ad90722 100644
--- a/readme.md
+++ b/readme.md
@@ -59,13 +59,13 @@ argument or look at the container via cat.
 * `REPORT_STATS`: statistic report, mandatory, values: `yes` or `no`, needed
   only for `generate`

-# build specific arguments
+# build specific variables

 * `BV_SYN`: synapse version, optional, defaults to `master`
 * `BV_TUR`: coturn turnserver version, optional, defaults to `master`

 For building of synapse version v0.11.0-rc2 and coturn with commit a9fc47e add
-`--build-arg BV_SYN=v0.11.0-rc2 --build-arg BV_TUR=a9fc47efd77` to the `docker
+`-e BV_SYN=v0.11.0-rc2 -e BV_TUR=a9fc47efd77` to the `docker
 build` command.

 # diff between system and fresh generated config file
```

**Pull request closed without merging**